src/containers/genePage/alleleTable.jsx

```diff
@@ -31,10 +48,80 @@ const AlleleTable = ({ isLoadingGene, gene, geneId }) => {
       },
       disease: allele.diseases?.sort(compareAlphabeticalCaseInsensitive((disease) => disease.name)),
     }));
+
+    // Filter to only show the selected alleles when in override mode
+    if (selectionOverride.active && selectionOverride.alleleIds.length > 0) {
+      processedData = processedData.filter((allele) => selectionOverride.alleleIds.includes(allele.id));
+    }
+
+    return processedData;
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [resolvedData]);
+  }, [resolvedData, selectionOverride, selectedAllelesData]);
 
-  const [alleleIdsSelected, setAlleleIdsSelected] = useState([]);
+  // Handler for allele selection that tracks source (table vs viewer)
+  const handleAllelesSelect = React.useCallback(
```

Comment:
I wonder if this function would be better as a hook. We usually have our code that calls the API as a hook. Maybe not something that needs to hold up the release now as long as it's working, but perhaps as an improvement for the future.

Reply:
I'm working on this now, I think I can actually convert it without too much trouble.